### PR TITLE
OCPBUGS-43657: audit: do not log requests to /livez

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
/assign @deads2k

for approval and adding `backport-risk-assessed` label

**Comment by wangke19** [comment]
Pre-merge verified the PR, got the expected results, detail see https://issues.redhat.com/browse/OCPBUGS-43657?focusedId=25929052&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25929052

**Comment by p0lyn0mial** [comment]
@liouk is is safe to overwrite failing `e2e-operator` ?

File: pkg/controllers/customroute/custom_route_conditions.go [hunk 1]
```diff
@@ -15,105 +15,123 @@ import (
...
 
-func ensureDefaultConditions(conditions []metav1.Condition) []metav1.Condition {
+var (
+	conditionStatusTrue      = metav1.ConditionTrue
+	conditionStatusFalse     = metav1.ConditionFalse
+	conditionTypeDegraded    = "Degraded"
+	conditionTypeProgressing = "Progressing"
+)
+
+func ensureDefaultConditions(conditions []*v1.ConditionApplyConfiguration) []*v1.ConditionApplyConfiguration {
```
Comment (deads2k) [inline pkg/controllers/customroute/custom_route_conditions.go:29]: how did we start using SSA in 4.17?

Comment (p0lyn0mial) [reply pkg/controllers/customroute/custom_route_conditions.go:29]: Note that this controller [is](https://github.com/openshift/cluster-authentication-operator/blob/release-4.17/pkg/controllers/customroute/custom_route_controller.go#L266) already using SSA in 4.17.

What changed is the signature of the `ComponentRouteStatusApplyConfiguration.WithConditions` method due to  https://github.com/openshift/client-go/pull/277

basically, before the change the method accepted:
```
(b *ComponentRouteStatusApplyConfiguration) WithConditions(values ...metav1.Condition) *ComponentRouteStatusApplyConfiguration
```

and after the change it accepts:
```
(b *ComponentRouteStatusApplyConfiguration) WithConditions(values ...*metav1.ConditionApplyConfiguration) *ComponentRouteStatusApplyConfiguration
```

The changes made in this PR also adapt the controller to use the new signature.

**Pull request merged**